### PR TITLE
Bring back: "Batching supports async model load. (#1540)

### PR DESCRIPTION
**Pull request by vkalpias**
This reverts commit 42c8c9f4cf5cee66db4194e22e4e403ad784f4b7.

Fixes #

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Hey @vkalpias, what would you like to do with this PR. Another old one.

File: src/framework/components/model/component.js [hunk 1]
```diff
@@ -267,7 +267,10 @@ Object.assign(pc, function () {
             }
 
             if (this._batchGroupId >= 0) {
-                app.batcher.insert(pc.BatchGroup.MODEL, this.batchGroupId, this.entity);
+                if (this._model)
+                    app.batcher.insert(pc.BatchGroup.MODEL, this.batchGroupId, this.entity);
+                else
```
Comment (Maksims) [inline src/framework/components/model/component.js:272]: Sorry for being picky, but please add `{ }` to match rest of the code conventions.

**Comment by vkalpias** [comment]
> Hey @vkalpias, what would you like to do with this PR. Another old one.

This is very old and I don't remember what this is supposed to be doing. I would lean towards closing this it was probably meant to fix a bug since it was for reverting another commit but if it wasn't merged I would assume it's not needed at this point.

**Comment by willeastcott** [comment]
OK, closing. 😄

**Pull request closed without merging**